### PR TITLE
Implement "glow in the dark" TS animations

### LINKED ISSUE #13046: TS building light animations need to glow in the dark
All building light overlays in the original were unaffected by map lighting, so we need to define a separate palette for them that won't get modified by global lighting and such.

### PR DESCRIPTION
**Pull request by pchote**
https://user-images.githubusercontent.com/167819/107078761-b151d000-67e6-11eb-9422-60f353375c0d.mov

This PR implements the "bright" palette ranges that were hardcoded for infantry and certain building animations in TS, as well as fixing many small errors in the sequence definitions. Best tested on a relatively dark map like Sunstroke, and the map editor can be used to quickly place buildings and check animations in normal and damaged states (muzzle flashes need to be tested in a real game).

Most animations that use this feature contain a mix of bright and non-bright colours, which I handle by creating new `player-nobright` and `player-nomuzzle` palettes that blank out the bright ranges, and corresponding `bright` and `muzzle` palettes that blank out everything else. Actors then render the animation twice, once with the `player-no*` and then again with `bright` or `muzzle` together with `IgnoreWorldTint: True`. A small number of animations only use the bright colours, so only specify the `-bright` version.

I'm fairly confident that this covers all of the animations that we currently define because I started this process by modifying the `CheckMissingSprites` utility command to report all sprites that used the bright colours, and only crossed animations off the list after I had verified them working ingame.

Fixes #13046.
~~Depends on #19122.~~

There are two other issues that aren't strictly blockers, but would be nice to resolve first:
* Rename `WithIdleOverlay` to `WithOverlay` and implement a proper `WithIdleOverlay` trait (including random `Interval` support) to properly implement the original refinery flame behaviour.
* Remove the `Cloak` palette override and reimplement using alpha: the stealth generator has always broken a few animations (like the refinery flame animation), but now breaks many more.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs [hunk 1]
```diff
@@ -35,13 +35,25 @@ public class WithSpriteBodyInfo : PausableConditionalTraitInfo, IRenderActorPrev
 		[Desc("Forces sprite body to be rendered on ground regardless of actor altitude (for example for custom shadow sprites).")]
 		public readonly bool ForceToGround = false;
 
+		[PaletteReference(nameof(IsPlayerPalette))]
+		[Desc("Custom palette name")]
+		public readonly string Palette = null;
+
+		[Desc("Palette is a player palette BaseName")]
```
Comment (teinarss) [inline OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs]: missing dot at the end?

File: OpenRA.Mods.Common/Traits/Render/WithMakeOverlay.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	[Desc("Draws an overlay on top of a make animation.")]
+	public class WithMakeOverlayInfo : TraitInfo
+	{
+		[SequenceReference]
+		[FieldLoader.Require]
+		[Desc("Sequence name to use.")]
+		public readonly string Sequence = null;
+
+		[PaletteReference(nameof(IsPlayerPalette))]
+		[Desc("Custom palette name")]
```
Comment (teinarss) [inline OpenRA.Mods.Common/Traits/Render/WithMakeOverlay.cs]: And here

File: OpenRA.Mods.Common/Traits/Render/WithMakeOverlay.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		public readonly string Palette = null;
+
+		[Desc("Custom palette is a player palette BaseName")]
```
Comment (teinarss) [inline OpenRA.Mods.Common/Traits/Render/WithMakeOverlay.cs]: !

File: OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs [hunk 4]
```diff
@@ -35,13 +35,25 @@ public class WithSpriteBodyInfo : PausableConditionalTraitInfo, IRenderActorPrev
 		[Desc("Forces sprite body to be rendered on ground regardless of actor altitude (for example for custom shadow sprites).")]
 		public readonly bool ForceToGround = false;
 
+		[PaletteReference(nameof(IsPlayerPalette))]
+		[Desc("Custom palette name")]
```
Comment (teinarss) [inline OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs]: !

**Comment by pchote** [comment]
Fixed those and the issues reported by the linter.

**Comment by pchote** [comment]
Rebased and ready for review.

**Comment by teinarss** [comment]
Mobile sensor array is missing its lighting animation when deployed.

**Comment by pchote** [comment]
Good spotting!

> I'm fairly confident that this covers all of the animations that we currently define

It turns out that this animation was never defined, so wasn't flagged by my automated reporting. I added a new commit to fix this and also fix the graphical glitches on snow maps.

**Comment by pchote** [comment]
Rebased for good measure.

**Review by teinarss — APPROVED** [review]

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/26a3e4a6a14271bd5fb8285d2b98688d3c093766...ba94c7de57f9052d1f3fd8556cf32103e79716ac

**Pull request merged**